src/bokeh_fastapi/application.py

```diff
@@ -0,0 +1,313 @@
+from __future__ import annotations
+
+import logging
+
+from typing import TYPE_CHECKING, Any, Mapping, Sequence
+from urllib.parse import urljoin
+
+from bokeh.application import Application
+from bokeh.application.handlers.document_lifecycle import DocumentLifecycleHandler
+from bokeh.application.handlers.function import FunctionHandler
+from bokeh.embed.bundle import extension_dirs
+from bokeh.resources import Resources
+from bokeh.server.connection import ServerConnection
+from bokeh.server.contexts import ApplicationContext
+from bokeh.server.session import ServerSession
+from bokeh.server.views.static_handler import StaticHandler
+from bokeh.settings import settings
+from fastapi import APIRouter, FastAPI
+from fastapi.staticfiles import StaticFiles
+
+from .handler import DocHandler, WSHandler
+
+if TYPE_CHECKING:
+    from bokeh.application.handlers.function import ModifyDoc
+
+    from bokeh.protocol import Protocol
+
+#-----------------------------------------------------------------------------
+# Globals and constants
+#-----------------------------------------------------------------------------
+
+log = logging.getLogger(__name__)
+
+DEFAULT_CHECK_UNUSED_MS                  = 17000
+DEFAULT_KEEP_ALIVE_MS                    = 37000  # heroku, nginx default to 60s timeout, so use less than that
+DEFAULT_UNUSED_LIFETIME_MS               = 15000
+DEFAULT_SESSION_TOKEN_EXPIRATION         = 300
+
+__all__ = (
+    'BokehFastAPI',
+)
+
+#-----------------------------------------------------------------------------
+# General API
+#-----------------------------------------------------------------------------
+
+class BokehFastAPI:
+    """
+        applications (dict[str,Application] or Application) :
+            A map from paths to ``Application`` instances.
+
+            If the value is a single Application, then the following mapping
+            is generated:
+
+            .. code-block:: python
+
+                applications = {{ '/' : applications }}
+
+            When a connection comes in to a given path, the associate
+            Application is used to generate a new document for the session.
+
+        prefix (str, optional) :
+            A URL prefix to use for all Bokeh server paths. (default: None)
+
+        secret_key (str, optional) :
+            A secret key for signing session IDs.
+
+            Defaults to the current value of the environment variable
+            ``BOKEH_SECRET_KEY``
+
+        sign_sessions (bool, optional) :
+            Whether to cryptographically sign session IDs
+
+            Defaults to the current value of the environment variable
+            ``BOKEH_SIGN_SESSIONS``. If ``True``, then ``secret_key`` must
+            also be provided (either via environment setting or passed as
+            a parameter value)
+
+        keep_alive_milliseconds (int, optional) :
+            Number of milliseconds between keep-alive pings
+            (default: {DEFAULT_KEEP_ALIVE_MS})
```

Comment:
This docstring is not used by anything, but without some other machinery, this variable will never be substituted.

Reply:
Adding keep-alive pings will definitely be needed, so we should crib from bokeh's implementation.

Reply:
I meant, the substitution in the docstring is not going to work. I assume bokeh has something for that in their doc builds, but without it, you'll get the raw string as written, which is not helpful.

But this is nitpicking. Not blocking in any sense.

Reply:
Ah, got it.